src/forms/FormControl.scss

```diff
@@ -104,166 +138,197 @@
 
-      .FormControl--input-trailingAction {
-        grid-column: 3 / 4;
+    &::after {
+      @include minTouchTarget(var(--primer-control-medium-size), var(--primer-control-medium-size));
+
+      @media (pointer: coarse) {
+        min-width: var(--primer-control-minTarget-coarse);
+        min-height: var(--primer-control-minTarget-coarse);
       }
     }
   }
 
-  // if trailingAction is present
-  &.FormControl-fieldWrap--input-trailingAction {
-    grid-template-columns: minmax(0, auto) min-content;
-  }
-
-  &.FormControl-fieldWrap--input {
-    grid-template-rows: max-content;
-
-    .FormControl--input-leadingVisual {
-      grid-column: 1 / 2;
-      grid-row: 1;
+  // if leadingVisual is present
+  &.FormControl-fieldWrap--input-leadingVisual {
+    .FormControl--input {
+      padding-inline-start:
+        calc(
+          var(--primer-control-medium-paddingInline-condensed, 8px) + var(--base-size-16, 16px) +
+          var(--primer-control-medium-gap, 8px) - var(--primer-borderWidth-thin, 1px)
+        );
     }
+  }
 
-    // <input> spans entire grid
+  // if trailingAction is present
+  &.FormControl-fieldWrap--input-trailingAction {
     .FormControl--input {
-      grid-column: 1 / -1;
-      grid-row: 1;
+      padding-inline-end:
+        calc(
+          var(--primer-control-medium-paddingInline-condensed, 8px) + var(--base-size-16, 16px) +
+          var(--primer-control-medium-gap, 8px) - var(--primer-borderWidth-thin, 1px)
+        );
     }
+  }
 
-    // TODO: replace with new Button component
-    // trailingAction will auto fill last grid slot
-    .FormControl--input-trailingAction {
-      display: grid;
-      width: calc(var(--primer-control-medium-size, 32px) - var(--base-size-4, 4px));
-      height: calc(var(--primer-control-medium-size, 32px) - var(--base-size-4, 4px));
-      padding: 0;
-      // optically align the icon with field padding
-      margin-right: calc(calc(var(--primer-control-medium-paddingInline-condensed, 8px) - 0.125rem) * -1);
-      color: var(--color-fg-muted);
-      cursor: pointer;
-      user-select: none;
-      background-color: transparent;
-      border: solid var(--primer-borderWidth-thin, 1px) transparent;
-      border-radius: var(--primer-borderRadius-medium, 6px);
-      transition: 0.2s cubic-bezier(0.3, 0, 0.5, 1);
-      transition-property: color, background-color, border-color;
-      grid-row: 1;
-      place-content: center;
-      place-self: center end;
-      grid-column: 2 / 3;
-
-      &:hover {
-        background-color: var(--color-btn-hover-bg);
-        border: solid var(--primer-borderWidth-thin, 1px) var(--color-btn-hover-bg);
+  // size modifications can be refactored with :has() - FormControl-fieldWrap:has(.FormControl--large)
+  // sizes
+  &.FormControl-fieldWrap--input {
+    &.FormControl--small {
+      .FormControl--input-leadingVisualWrap {
+        top: calc(var(--primer-control-medium-paddingInline-condensed, 8px) - 0.125rem); /* 6px */
```

Comment:
I don't love that we have to do a `calc` to get a spacing value that works, but this is _way_ better than hard-coding `6px`.

Is there any significance to the number `0.125`? Or is it just what worked best?

Reply:
Would we ever use the 2px sizing token on it's own? Or do you mostly find yourself reaching for it in `calc()`?

If it's mostly just for `calc()`, then I think it's best to keep using relative units.